recipe/meta.yaml

```diff
@@ -1,3 +1,25 @@
+# This recipe creates 3 packages
+# libopencv, py-opencv, opencv
+#
+# libopencv actually contains all the files, including the site-packages
+# file needed to run opencv with python.
+#     opencv-dnn seems to change on every patch version, therefore, we pin
+#     to the patch version x.x.x
+#
+# py-opencv is a meta package that constrains the python dependencies
+#     This one is pinned to the minor version. x.x
+#
+# opencv is a backward compatible package, essentially the same as
+#     py-opencv but pinned to x.x.x
+#
+# This approach is different than the one in defaults. Default's package
+# seperates well the files generated with python dependencies, from those
+# generated without. Unfortunately, this isn't entirely supported by
+# upstream. Their hacks are quite elaborate, making it seem difficult to
+# maintain as opencv has been changing their build system recently.
```

Comment:
Can we not add a script to the libopencv output to remove the python or site-packages directory? From the help https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#outputs-section:

> For scripts that move or create files, a fresh copy of the working directory is provided at the start of each script execution. This ensures that results between scripts are independent of one another.

So, I presume this means we are safe to delete the unwanted files?

Reply:
Honestly, I really don't think it is worth it.

I remember trying that a while back, but for some reason, "deleting" files after they had been moved in didn't work.

You are free to try, but I don't have bandwidth to do that fix seeing as it solves the dependency problem, without adding too much complication to the script now.
For example, I updated the recipe (needs to be rebased now) for 4.0.0 and some reorg already made it a slightly more involved update. https://github.com/conda-forge/opencv-feedstock/pull/138/files

In my experience with split recipes, you often had to rerun the `make install` command.
But that command will recheck the environment, and if any changes are detected, it will rebuild.

Therefore, you have to relist all the build dependencies. For open cv, this is an extra 10-20 lines that we have to remember not to have typos in, or make mistakes in selectors/pinnings.

We should really be spending our effort asking opencv to make a sensible python binding, that doesn't require ALL the libraries to exist. Allowing us to compile with all the optional libraries.

This is what defaults tried to do:
install python script: 60 lines of code https://github.com/AnacondaRecipes/opencv-feedstock/blob/master/recipe/install-py-opencv.sh
meta file: 240 lines https://github.com/AnacondaRecipes/opencv-feedstock/blob/master/recipe/meta.yaml

This recipe keeps avoid the install python script, and the meta is about 200 if you remove my big comment at the front.

Reply:
Fair enough, thanks for the detailed explanation. I don't have the bandwidth to work on this, but suffice to say, it would be nice if conda-build supported some of this capability with their outputs mechanism so we could have fewer hacks in the recipes and built packages.